spec/mysql2/client_spec.rb

```diff
@@ -288,6 +288,14 @@ def run_gc
     }.to raise_error(Mysql2::Error)
   end
 
+  it "should not try to query closed mysql connection" do
```

Comment:
This test is correct. If the current behavior is not respecting that the connection has been closed, we may consider adding an extra flag or additional state/value for the `wrapper->connected` field.